homeassistant/components/starline/api.py

```diff
@@ -0,0 +1,260 @@
+"""StarLine API."""
+import aiohttp
+import hashlib
+
+from datetime import timedelta
+from typing import Dict, List, Callable, Optional, Any
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.event import async_track_time_interval
+
+from .device import StarlineDevice
+from .const import (
+    LOGGER,
+    ENCODING,
+    GET,
+    POST,
+    CONNECT_TIMEOUT,
+    TOTAL_TIMEOUT,
+    DEFAULT_UPDATE_INTERVAL,
+)
+
+
+class BaseApi:
```

Comment:
Please extract the device specific interface code, I/O and data parsing, to a standalone library published on pypi.

https://developers.home-assistant.io/docs/en/creating_component_code_review.html#4-communication-with-devices-services

Reply:
Done https://pypi.org/project/starline/